Sources/sourcekit-lsp/main.swift

```diff
@@ -161,8 +161,8 @@ struct Main: ParsableCommand {
 
     let clientConnection = JSONRPCConnection(
       protocol: MessageRegistry.lspProtocol,
-      inFD: fileno(stdin),
-      outFD: realStdout,
+      inFD: FileHandle(fileDescriptor: fileno(stdin), closeOnDealloc: false),
+      outFD: FileHandle.standardInput,
```

Comment:
You changed the wrong file handle; should be something like this, I think:

```swift
inFD: FileHandle.standardInput,
outFD: FileHandle(fileDescriptor: realStdout, closeOnDealloc: false),
```

In its current state it's not likely to behave very well 🤣

Reply:
Nice!  Totally a force merge candidate :-p